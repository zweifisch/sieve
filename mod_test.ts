import {
  assertSnapshot,
} from "https://deno.land/std@0.216.0/testing/snapshot.ts";
import {
  assertEquals,
} from "https://deno.land/std@0.216.0/testing/asserts.ts"
import { SieveCache } from "./mod.ts"


Deno.test("should evict", async (t) => {

  const cache = new SieveCache<string>(3)
  cache.setItem('A', 'a')
  cache.setItem('B', 'b')
  cache.setItem('C', 'c')

  await assertSnapshot(t, cache.toString())

  cache.setItem('D', 'd')

  await assertSnapshot(t, cache.toString())

  // cache.getItem('B')
  // cache.setItem('E', 'e')


})
