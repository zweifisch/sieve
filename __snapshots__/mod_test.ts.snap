export const snapshot = {};

snapshot[`should evict 1`] = `
[32m"[ ] C
[ ] B
[ ] A"[39m
`;

snapshot[`should evict 2`] = `
[32m"[ ] D
[ ] C
[ ] B"[39m
`;
